packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -405,14 +465,12 @@ export async function generateBlogPosts(
     await Promise.all(blogSourceFiles.map(doProcessBlogSourceFile))
   ).filter(Boolean) as BlogPost[];
 
-  blogPosts.sort(
-    (a, b) => b.metadata.date.getTime() - a.metadata.date.getTime(),
-  );
+  const sortedPosts = sortBlogPosts({
+    blogPosts,
+    sortPosts: options.sortPosts,
+  });
 
-  if (options.sortPosts === 'ascending') {
-    return blogPosts.reverse();
-  }
-  return blogPosts;
+  return sortedPosts;
```

Comment:
There is no blog sorting by default now which cause doesn't cause problem for the site but some tests are failing

Wondering if the default value should be given like this :
```ts
  const sortedPosts = sortBlogPosts({
    blogPosts,
    sortPosts: options.sortPosts ?? 'descending',
  });
  ```
or if it should be in the function declaration like this :
```ts
function sortBlogPosts({
  blogPosts,
  sortPosts = 'descending',
})
```

Reply:
The problem is not related to the default value, but with your assumption that the sort function always returns a list of sorted blog posts.

By yes we should keep it retrocompatible and sort descending by default (in the schema default values, not in code)

Our API allows users to provide `sortBlogPosts: () => {}`. Returning undefined is possible.


so when you do `return sortedBlogPosts` you might return undefined

Reply:
Looks like the feed tests have been fixed now. Since the option has a default value, this option should be provided in all tests to reflect real usage.

The feeds tests worked before because the code was more defensive and performed the sort even when `sortPosts` was undefined. But in practice the default value ensures `sortPosts` is never undefined so we shouldn't need to program defensively against that possibility.

Resolved 👍